src/lakefs_spec/spec.py

```diff
@@ -33,6 +33,30 @@
 EmptyYield = Generator[None, None, None]
 
 
+class LakectlConfig(NamedTuple):
+    host: str | None = None
+    username: str | None = None
+    password: str | None = None
+
+    @classmethod
+    def read(cls, path: str | Path) -> "LakectlConfig":
+        try:
+            import yaml
+        except ModuleNotFoundError:
+            return cls()
+
+        obj: dict[str, Any] = yaml.safe_load(path)
```

Comment:
This throws the following error for me

```
File "/<filepath>/stuff.py", line 4, in <module>
    obj = yaml.safe_load(Path('~/.lakectl.yaml'))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<filepath>/.venv/lib/python3.11/site-packages/yaml/__init__.py", line 125, in safe_load
    return load(stream, SafeLoader)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<filepath>/.venv/lib/python3.11/site-packages/yaml/__init__.py", line 79, in load
    loader = Loader(stream)
             ^^^^^^^^^^^^^^
  File "/<filepath>/.venv/lib/python3.11/site-packages/yaml/loader.py", line 34, in __init__
    Reader.__init__(self, stream)
  File "/<filepath>/.venv/lib/python3.11/site-packages/yaml/reader.py", line 85, in __init__
    self.determine_encoding()
  File "/<filepath>/.venv/lib/python3.11/site-packages/yaml/reader.py", line 124, in determine_encoding
    self.update_raw()
  File "/<filepath>/.venv/lib/python3.11/site-packages/yaml/reader.py", line 178, in update_raw
    data = self.stream.read(size)
           ^^^^^^^^^^^^^^^^
AttributeError: 'PosixPath' object has no attribute 'read'
```

Which originates from the `PyYAML` package. Does it work for you?

Reply:
Does not occur with your recent changed. Marking as resolved.

Reply:
Already fixed at HEAD, this is exactly the content of [e3d4d0b](https://github.com/appliedAI-Initiative/lakefs-spec/pull/45/commits/e3d4d0b60226fb856412fb7fde7f269b8a3fc4fc).